dev/buildtool/git.py

```diff
@@ -490,12 +551,68 @@ def __normalize_repo_url(url):
     return os.path.abspath(url)
 
   @staticmethod
-  def same_repo(first, second):
+  def is_same_repo(first, second):
     """Determine if two URLs refer to the same github repo."""
     normalized_first = GitRunner.__normalize_repo_url(first)
     normalized_second = GitRunner.__normalize_repo_url(second)
     return normalized_first == normalized_second
 
+  @property
+  def options(self):
+    """Return bound options."""
+    return self.__options
+
+  def __init__(self, options):
+    self.__options = options
+    self.__auth_env = {}
+    if self.__options.github_token_path:
```

Comment:
I know we've done it this way in the past, but storing tokens in plaintext files is not a good pattern. I'm not sure about the other ramifications of opting for an env var over a file (like implicitly providing subprocesses with the git credentials per your comment), but we may want to rethink using files for this.

Reply:
Ok. I agree. I have a plan B. Hold on...

Reply:
ok. have a look. I can disregard the whole special casing if not.